tests/test_models.py

```diff
@@ -755,6 +755,108 @@ def test_answers_can_be_100_words(self):
+            ],
+            'description': u'',
+            'id': 0,
+            'links': {'self': 'http://127.0.0.1:5000/suppliers/0'},
+            'name': u'Supplier 0',
+        }
+
+    def test_update_from_json(self):
+        with self.app.app_context():
+            initial_id = self.supplier.id
+            initial_sid = self.supplier.supplier_id
+
+            self._update_supplier_from_json_with_all_details()
+
+            # Check IDs can't be updated
+            assert self.supplier.id == initial_id
+            assert self.supplier.supplier_id == initial_sid
+
+            # Check everything else has been updated to the correct value
+            assert self.supplier.name == "String and Sticky Tape Inc."
+            assert self.supplier.clients == ["Parcel Wrappers Ltd"]
+            assert self.supplier.duns_number == "01010101"
+            assert self.supplier.esourcing_id == "020202"
+            assert self.supplier.description == "All your parcel wrapping needs catered for"
+            assert self.supplier.companies_house_number == "98765432"
+            assert self.supplier.registered_name == "Tape and String Inc."
+            assert self.supplier.registration_country == "Wales"
+            assert self.supplier.other_company_registration_number == ""
+            assert self.supplier.registration_date == datetime(1973, 8, 10, 0, 0)
+            assert self.supplier.vat_number == "321321321"
+            assert self.supplier.organisation_size == "medium"
+            assert self.supplier.trading_status == "Sticky"
+
+    def test_serialization_of_supplier_with_all_details_added(self):
+        with self.app.app_context():
+            self._update_supplier_from_json_with_all_details()
+            assert self.supplier.serialize() == {
+                'clients': ['Parcel Wrappers Ltd'],
+                'companiesHouseNumber': '98765432',
+                'contactInformation': [
+                    {
+                        'contactName': u'Contact for Supplier 0',
+                        'email': u'0@contact.com',
+                        'id': 13,
+                        'links': {'self': 'http://127.0.0.1:5000/suppliers/0/contact-information/13'},
+                        'postcode': u'SW1A 1AA',
+                    }
+                ],
+                'description': 'All your parcel wrapping needs catered for',
+                'dunsNumber': '01010101',
+                'eSourcingId': '020202',
+                'id': 0,
+                'links': {'self': 'http://127.0.0.1:5000/suppliers/0'},
+                'name': 'String and Sticky Tape Inc.',
+                'organisationSize': 'medium',
+                'otherCompanyRegistrationNumber': '',
+                'registeredName': 'Tape and String Inc.',
+                'registrationCountry': 'Wales',
+                'registrationDate': '1973-08-10',
+                'tradingStatus': 'Sticky',
+                'vatNumber': '321321321',
+            }
```

Comment:
Hmm, my thinking when I wrote this was that the first tests the update_from_json method and the second ensures things are serialized as we expect, but in fact the serialization test depends on the update_from_json working properly, so yep maybe they should be assertions in the same test.  And like you say, those milliseconds are precious.